### PR TITLE
Check primary ctor in UndocumentedPublicProperty

### LINKED ISSUE #2468: UndocumentedPublicProperty doesn't check primary constructor
## Expected Behavior
If a property is declared in the primary constructor (especially in a data class), the static analysis should count an issue if the property isn't documented. 

## Observed Behavior
Only properties declared in the class body are flagged when left public and undocumented.

## Steps to Reproduce
Create a class with properties declared in the primary constructor but don't  add `@property` or KDoc reference in a comment

## Your Environment

* Version of detekt used: 1.6.0
* Operating System and version: OSX

### PR DESCRIPTION
**Pull request by schalkms**
Closes #2468

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt [hunk 1]
```diff
@@ -90,13 +100,34 @@ class UndocumentedPublicPropertySpec : Spek({
...
+        it("does not report undocumented public properties in a primary constructor for an internal class") {
+            val code = "internal class Test(val a: Int)"
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("does not report documented public properties in a primary constructor") {
+            val code = """
+                /**
+                * @a int1
+                * [b] int2
```
Comment (BraisGabin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt]: This doesn't seem like kdoc: https://kotlinlang.org/docs/reference/kotlin-doc.html 
Shouldn't it be `@property a`?

Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt]: This syntax is used for linking. Please take a look at the section on the same page.
https://kotlinlang.org/docs/reference/kotlin-doc.html#linking-to-elements

Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt]: The sample on the top of the page also shows how to use the linking syntax.
https://kotlinlang.org/docs/reference/kotlin-doc.html#kdoc-syntax

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt [hunk 2]
```diff
@@ -90,13 +100,34 @@ class UndocumentedPublicPropertySpec : Spek({
...
-                    constructor(a: Int, b: Int) : this(a)
+                class Test() {
+                    constructor(a: Int) : this()
                 }
             """
             assertThat(subject.compileAndLint(code)).isEmpty()
         }
+
+        it("does not report undocumented non-public properties in a primary constructor") {
+            val code = "class Test(internal val a: Int)"
```
Comment (t-kameyama) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt]: Please consider following case. I think it should not be reported. 
```
class Test(a: Int)
```

Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt]: Thanks for your review and for your feedback.
I added a test case for this snippet.

**Pull request merged**